### PR TITLE
[flutter_markdown] Unable to use MarkdownElementBuilder to act those tags without children.

### PR DESCRIPTION
**Pull request by MeandNi**
fix https://github.com/flutter/flutter/issues/126402

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [  ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [  ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [  ] I updated/added relevant documentation (doc comments with `///`).
- [  ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [  ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by MeandNi** [comment]
> Please add tests that exercise the error condition that this PR fixes↳

done

**Comment by domesticmouse** [comment]
Looking at CI, looks like there are some issues:

```

Running command: "dart analyze --fatal-infos" in /tmp/cirrus-ci-build/packages/flutter_markdown
Analyzing flutter_markdown...
   info - lib/src/builder.dart:457:15 - Use 'isEmpty' instead of 'length' to test whether the collection is empty. Try rewriting the expression to use 'isEmpty'. - prefer_is_empty
   info - test/custom_syntax_test.dart:136:31 - Unnecessary use of double quotes. Try using single quotes unless the string contains single quotes. - prefer_single_quotes
   info - test/custom_syntax_test.dart:256:17 - Unnecessary use of double quotes. Try using single quotes unless the string contains single quotes. - prefer_single_quotes
3 issues found.
```

**Comment by domesticmouse** [comment]
Cool! Can you now please update the version and add some notes to the changedoc about this change? It won't be released to the world without bumping the version =)

**Review by domesticmouse — APPROVED** [review]
PTAL @stuartmorgan

File: packages/flutter_markdown/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-## NEXT
+## 0.6.15
 
 * Updates minimum Flutter version to 3.3.
 * Aligns Dart and Flutter SDK constraints.
 * Replace `describeEnum` with the `name` getter.
+* Support custom rendering of those tags without children.
```
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:6]: Please follow the style in [the guide linked from the PR checklist above](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style)

Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:6]: Remove "those"

File: packages/flutter_markdown/test/custom_syntax_test.dart [hunk 2]
```diff
@@ -112,6 +112,30 @@ void defineTests() {
       expect(widgetSpan.child, isInstanceOf<Container>());
     },
   );
+
+  testWidgets(
+    'Custom rendering of those tags without children',
```
Comment (stuartmorgan) [inline packages/flutter_markdown/test/custom_syntax_test.dart:117]: Remove "those"

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**